extensions/amp-skimlinks/0.1/tracking.js

```diff
@@ -0,0 +1,284 @@
+      data: JSON.stringify(data),
+      rnd: 'RANDOM',
+    });
+  }
+
+  /**
+   * Sends tracking request to Skimlinks tracking API in order to
+   * register page impression request.
+   * @private
+   * @param {Object} commonData
+   * @param {number} numberAffiliateLinks
+   */
+  sendPageImpressionTracking_(commonData, numberAffiliateLinks) {
+    const {customTrackingId, referrer} = this.trackingInfo_;
+
+    const data = /** @type {!JsonObject} */ (Object.assign(
+        {
+          slc: numberAffiliateLinks,
+          // Javascript load time, not relevent in AMP context.
+          jsl: 0,
+          pref: referrer,
+          uc: customTrackingId,
+          t: 1,
+        },
+        commonData
+    ));
+
+    // Sends POST request. Second param is the object used to interpolate
+    // placeholder variables defined in PAGE_IMPRESSION_TRACKING_URL.
+    this.analytics_.trigger('page-impressions', {
+      data: JSON.stringify(data),
+    });
+  }
+
+  /**
+   * Sends tracking request to Skimlinks tracking API in order to
+   * register link impression request.
+   * @private
+   * @param {Object} commonData
+   * @param {number} numberAffiliateLinks
+   * @param {Object} urls
+   */
+  sendLinkImpressionTracking_(commonData, numberAffiliateLinks, urls) {
+    const data = /** @type {!JsonObject} */ (Object.assign(
+        {
+          dl: urls,
+          hae: numberAffiliateLinks ? 1 : 0, // 1 if has at least one AE link
+          typ: 'l',
+        },
+        commonData
+    ));
+
+    // Send POST request. Second param is the object used to interpolate
+    // placeholder variables defined in LINKS_IMPRESSIONS_TRACKING_URL.
+    this.analytics_.trigger('link-impressions', {
+      data: JSON.stringify(data),
+    });
+  }
+
+  /**
+   * Initialise the amp-analytics internal API.
+   * Warning: Analytics API will not send any request until
+   * CommonSignals.LOAD_START has been triggered.
+   * (See "initTracking_" in amp-skimlinks.js)
+   * @private
+   * @param {AmpElement} element
+   */
+  setupAnalytics_(element) {
+    const analyticsBuilder = new CustomEventReporterBuilder(element);
+    // Configure analytics to send POST request when receiving
+    // 'page-impressions' event.
+    analyticsBuilder.track('page-impressions', PAGE_IMPRESSION_TRACKING_URL);
+    analyticsBuilder.track('link-impressions', LINKS_IMPRESSIONS_TRACKING_URL);
+    analyticsBuilder.track('non-affiliate-click', NA_CLICK_TRACKING_URL);
+
+    const analytics = analyticsBuilder.build();
+    /*
+      Use {beacon: true} to send the request through sendBeacon()
+      https://www.ampproject.org/docs/analytics/deep_dive_analytics#how-data-gets-sent:-transport-attribute
+
+      Overwrite config manually since CustomEventReporterBuilder doesn't
+      support optional config.
+      TODO: add optional config param to .build() so we don't need to mutate
+      a private property from outside.
+    */
+    analytics.config_.transport = {beacon: true};
```

Comment:
This won't work, the access to private variable will be broken after compilation. 

@slocka To confirm: you only want to send request via `sendBeacon()`? what about the case when beacon is not supported? 
By default the transport config will be `{image: true, post: true, beacon: true}` will that work for you?

opened #19014